packages/grafana-ui/src/types/icon.ts

```diff
@@ -201,7 +201,7 @@ export const getAvailableIcons = () => availableIcons;
 
 type BrandIconNames = typeof avaibleBrandIcons;
 
-export type IconName = ReturnType<typeof getAvailableIcons>[number] | BrandIconNames[number];
+export type IconName = ReturnType<typeof getAvailableIcons>[number] | BrandIconNames[number] | string;
```

Comment:
Allowing a string as icon might enhance the problem of not-displaying icons since we don't have any way to check the icons we are using actually exists. 

I believe we need to have control of what we allow to be used as an Icon so we can check in a programmatic way the icon will actually display.

I had a PR where I was trying to introduce all the available icons inside this type https://github.com/grafana/grafana/pull/53820 but after conversations we decided opening the icons to every single possibility can result in an uneven usage of icons.